tests/test_dataset.py

```diff
@@ -14,23 +15,47 @@
 
 node_feats = [Nfeat.RESTYPE, Nfeat.POLARITY, Nfeat.BSA, Nfeat.RESDEPTH, Nfeat.HSE, Nfeat.INFOCONTENT, Nfeat.PSSM]
 
+def _cal_mean_std( # noqa: MC0001, pylint: disable=too-many-locals
+                       hdf5_path: str,
+                       features_transform:dict,
+                       feat:str
+    ):
+
+    with h5py.File(hdf5_path, 'r') as f:
+        entry_names = [entry for entry, _ in f.items()]
+
+        df_dict = {}
+        df_dict['id'] = entry_names
+
+        transform = False
+        transform = features_transform.get(feat, {}).get('transform')
+                
+        df_dict[feat] = [
+            f[entry_name][Nfeat.NODE][feat][:]
+            if f[entry_name][Nfeat.NODE][feat][()].ndim == 1
+            else f[entry_name][Nfeat.NODE][feat][()] for entry_name in entry_names]
+        #apply transformation
+        if transform:
+            df_dict[feat]=[transform(row) for row in df_dict[feat]]
+        
+        df = pd.DataFrame(data=df_dict)
```

Comment:
We don't need a pandas df here, since we're calculating the mean and the std dev for a single feature `feat`. You can safely remove `df_dict` and `df` variables. Instead, you can just calculate what now is `df_dict[feat]` as something like `feat_values` in the form of a list (same way as it is now) which then can be transformed if needed and finally converted into a numpy array for computing the mean and the std dev.